### PR TITLE
New Predefined Recognizer: IN_PAN

### PR DESCRIPTION
**Pull request by devopam**
## Change Description
Added presidio predefined recognizer : IN_PAN 

Describe your changes
Added pattern and context based recognizer InPanRecognizer to predefined recognizers. 
This recognizer is for India Permanent Account Number (PAN) which is the primary unique taxpayer identifier for all citizens.

## Issue reference

This PR fixes issue #XX

## Checklist

- [X] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [X] I have signed the CLA (if required)
- [X] My code includes unit tests
- [X] All unit tests and lint checks pass locally
- [X] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by devopam** [comment]
@microsoft-github-policy-service agree [company="{BlackhawkNetwork}"]

**Comment by devopam** [comment]
@microsoft-github-policy-service agree company="Blackhawk Network"

**Comment by omri374** [comment]
@devopam please run `black` and `flake8` to make sure there are no linting issues. Thanks!

**Comment by devopam** [comment]
that was a bad miss. fixed and ran lint - no errors reported on local. Also, black recommended no changes

File: presidio-analyzer/tests/test_in_pan_recognizer.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+
+@pytest.mark.parametrize(
+    "text, expected_len, expected_position, expected_score",
+    [
+        # fmt: off
+        ("AAASA1111R", 1, (0,10), 0.6) ,
+        ("ABCPD1234Z", 1, (0, 10), 0.85),
+        ("ABCND1234Z", 1, (0, 10), 0.6),
+        ("A1111DFSFS", 1, (0,10),0.05),
```
Comment (omri374) [inline presidio-analyzer/tests/test_in_pan_recognizer.py:24]: Can you please add a few negative examples?

Comment (devopam) [reply presidio-analyzer/tests/test_in_pan_recognizer.py:24]: hi @omri374 , can you please have a look at tests/data/context_sentences_test.txt (last 3 ) and let me know if those are sufficient.

Comment (omri374) [reply presidio-analyzer/tests/test_in_pan_recognizer.py:24]: Thanks. Please see a reference here: https://github.com/microsoft/presidio/blob/1b470616ab422ca145f438e96af969e9f18646d1/presidio-analyzer/tests/test_ip_recognizer.py#L23

There are cases where the input should not be detected, and the expected length is 0. Furthermore, there are cases where the input is part of a sentence (to check that the patterns are not limiting to full text patterns only)

**Review by omri374 — APPROVED** [review]
Thanks! LGTM

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/in_pan_recognizer.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+from presidio_analyzer import Pattern, PatternRecognizer
+
+
+class InPanRecognizer(PatternRecognizer):
+    """
+    Recognizes Indian Permanent Account Number ("PAN").
+
+    The Permanent Account Number (PAN) is a ten digit alpha-numeric code
+    with the last digit being a check digit calculated using a
```
Comment (SharonHart) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/in_pan_recognizer.py:11]: you can add that as part of a check-sum check under the ABC method of v'alidate_result'

**Review by SharonHart — APPROVED** [review]
Thanks for the contribution
see my comment about adding the check-sum functionality to make the recognizer even more robust.
Also add the new entity to the documentation

**Comment by omri374** [comment]
@devopam would you be interested in adding the checksum logic in a new PR?

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/in_pan_recognizer.py (hunk 2 above)
Comment (devopam) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/in_pan_recognizer.py:11]: As suggested by @omri374 , i will take this up in a new PR. This sounds like a valid addition to be explored. Makes the identification more robust by design.

**Comment by omri374** [comment]
Mergine this one, will be adding the checksum capability in a future PR.

**Pull request merged**